crypto-com/chain
# Problem: no address discovery after wallet recovery#1396

Why i resolved this issue?
1. because in hd wallet restoration, i didn't see tx history. there is no way to do it except for manual importing, which is very hard to use.
2. this feature enables very handy managing of private info as well, because user only need to remember this hdwallet mnemonics.
3. also this is quite working well with "sync with loop", balance is fetched right away automatically.

Solution:
1. added dynamic address recovery in syncing
2. used 20 addresses sliding window to discover new addresses,
if addresses are beyond that limit, it's regarded as not owned ones
3. there was sync glitch for tx history recovery, this is fixed, too.

Difficulty in working this issue:
after working dynamic address recovery, some txs are displayed as "incoming", because in syncing, input TxOut was not written to Wallet State,
which is pre-fetched in beginning of sync, which is quite old data. which is not updated while in processing each TX. 
(this will be done as another PR)
so that TX's input UTXO isn't identified, so system consider it as "incoming" (from the staking).
obvisously it's not, it's "outgoing" operation to other wallet.

How to check:
1. made transfer
2. remove .storage of client-rpc
3. restore hdwallet
4. sync with loop
5. you can see txhistory
6. make new wallet, repeat above
